iotc-generic-c-sdk/include/iotconnect.h

```diff
@@ -64,10 +64,12 @@ typedef struct {
     IotclCommandCallback cmd_cb; // callback for command events.
     IotclMessageCallback msg_cb; // callback for ALL messages, including the specific ones like cmd or ota callback.
     IotConnectStatusCallback status_cb; // callback for connection status
+    bool dynamic_memory;
```

Comment:
Since the local config is static in file scope, it should be a lot cleaner to have a static file scope variable is_config_cloned and drive the code based on that. That way the user des not get confused about "what is this dynamic_memory thing that I aparantly need to specify in config".